scripts/check_imports.swift

```diff
@@ -121,59 +127,65 @@ func checkFile(_ file: String, isPublic: Bool) {
-var url = URL(fileURLWithPath: FileManager().currentDirectoryPath)
-while url.path != "/", url.lastPathComponent != "firebase-ios-sdk" {
-  url = url.deletingLastPathComponent()
-}
-
-let repoURL = url
-
-let contents =
-  try FileManager.default.contentsOfDirectory(at: repoURL,
-                                              includingPropertiesForKeys: nil,
-                                              options: [.skipsHiddenFiles])
-
-for rootURL in contents {
-  if !rootURL.hasDirectoryPath {
-    continue
+private func main() {
+  // Search the path upwards to find the root of the firebase-ios-sdk repo.
+  var url = URL(fileURLWithPath: FileManager().currentDirectoryPath)
+  while url.path != "/", url.lastPathComponent != "firebase-ios-sdk" {
+    url = url.deletingLastPathComponent()
   }
-  let enumerator = FileManager.default.enumerator(atPath: rootURL.path)
-  whileLoop: while let file = enumerator?.nextObject() as? String {
-    if let fType = enumerator?.fileAttributes?[FileAttributeKey.type] as? FileAttributeType,
-      fType == .typeRegular {
-      if file.starts(with: ".") {
-        continue
-      }
-      if !(file.hasSuffix(".h") ||
-        file.hasSuffix(".m") ||
-        file.hasSuffix(".mm") ||
-        file.hasSuffix(".c")) {
-        continue
-      }
-      let fullTransformPath = rootURL.path + "/" + file
-      for dirPattern in skipDirPatterns {
-        if fullTransformPath.range(of: dirPattern) != nil {
-          continue whileLoop
+  let repoURL = url
+  guard let contents = try? FileManager.default.contentsOfDirectory(at: repoURL,
+                                                                    includingPropertiesForKeys: nil,
+                                                                    options: [.skipsHiddenFiles])
+  else {
+    ErrorLogger.log("Failed to get repo contents \(repoURL)")
+    return
+  }
+
+  for rootURL in contents {
+    if !rootURL.hasDirectoryPath {
+      continue
+    }
+    let enumerator = FileManager.default.enumerator(atPath: rootURL.path)
+    whileLoop: while let file = enumerator?.nextObject() as? String {
+      if let fType = enumerator?.fileAttributes?[FileAttributeKey.type] as? FileAttributeType,
+        fType == .typeRegular {
+        if file.starts(with: ".") {
+          continue
+        }
+        if !(file.hasSuffix(".h") ||
+          file.hasSuffix(".m") ||
+          file.hasSuffix(".mm") ||
+          file.hasSuffix(".c")) {
+          continue
+        }
+        let fullTransformPath = rootURL.path + "/" + file
+        for dirPattern in skipDirPatterns {
+          if fullTransformPath.range(of: dirPattern) != nil {
+            continue whileLoop
+          }
         }
+        checkFile(fullTransformPath, inRepo: repoURL, isPublic: file.range(of: "/Public/") != nil)
       }
-      checkFile(fullTransformPath, isPublic: file.range(of: "/Public/") != nil)
     }
   }
 }
 
-exit(foundError ? 1 : 0)
+main()
```

Comment:
perhaps have main return the status code?

Reply:
Done